### PR TITLE
Have the RemoteWorkspace work with the diff returned by unicore.distribute's pull.

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

**Comment by smn** [comment]
@Rizziepit & @miltontony ready for review. Lots of mocking going on because I don't want to have to implement the whole server side of things for this test.

File: elasticgit/tests/base.py [hunk 1]
```diff
@@ -80,6 +80,18 @@ def mk_workspace(self, working_dir=None,
 
         return workspace
 
+    def create_upstream_for(self, workspace, create_remote=True,
```
Comment (Rizziepit) [inline elasticgit/tests/base.py]: If it's called more than once in a test, it's likely going to create a duplicate workspace and possibly leave an index lying around. Just a heads up if this is likely to be used in more tests.

Comment (smn) [inline elasticgit/tests/base.py]: The workspaces should cleanup after themselves: https://github.com/universalcore/elastic-git/blob/develop/elasticgit/tests/base.py#L70-L71

Comment (Rizziepit) [inline elasticgit/tests/base.py]: The issue is that two calls to `create_upstream_for` in the same test will use the same workspace name. This means only the first `destroy` call does anything. The second sees the repo is gone and doesn't clean up the (possibly different) index.

(This is the silly test issue I ran into in #57.)

File: elasticgit/workspace.py [hunk 2]
```diff
@@ -344,14 +344,65 @@ def __init__(self, url, es=None, index_prefix=None):
...
+            else:
+                path_info = self.sm.path_info(change.get('path'))
+                if path_info is not None:
+                    changed_model_set.add(path_info[0])
+
+        for model_class in changed_model_set:
+            self.reindex(model_class)
+
     def pull(self, branch_name='master', remote_name='origin'):
         # TOOD: In the local storage we're diffing the changes pulled in
```
Comment (Rizziepit) [inline elasticgit/workspace.py]: This TODO can be removed I think

**Comment by Rizziepit** [comment]
Done reviewing. +1

Upgrading `mk_workspace`/`create_upstream_for` is a nice-to-have. If you think it's worthwhile, it can go in its own issue.

**Comment by miltontony** [comment]
Ran my eyes through this as well

File: elasticgit/tests/base.py (hunk 1 above)
Comment (smn) [inline elasticgit/tests/base.py]: Ah yes, then you have to provide a different `suffix` parameter. I'm inclined to leave it as is because this code's already being used in several places at the moment.
